src/client/Microsoft.Identity.Client/AuthScheme/PoP/PoPAuthenticationScheme.cs

```diff
@@ -65,31 +65,51 @@ public PoPAuthenticationScheme(PoPAuthenticationConfiguration popAuthenticationC
 
         public string FormatAccessToken(MsalAccessTokenCacheItem msalAccessTokenCacheItem)
         {
-            var header = new JObject
+            JObject header = new JObject
             {
                 { JsonWebTokenConstants.ReservedHeaderParameters.Algorithm, _popAuthenticationConfiguration.PopCryptoProvider.CryptographicAlgorithm },
                 { JsonWebTokenConstants.ReservedHeaderParameters.KeyId, KeyId },
                 { JsonWebTokenConstants.ReservedHeaderParameters.Type, PoPRequestParameters.PoPTokenType}
             };
 
-            JToken popAssertion = JToken.Parse(_popAuthenticationConfiguration.PopCryptoProvider.CannonicalPublicKeyJwk);
+            JObject body = CreateBody(msalAccessTokenCacheItem);
 
-            var payload = new JObject(
-                new JProperty(PoPClaimTypes.At, msalAccessTokenCacheItem.Secret),
-                new JProperty(PoPClaimTypes.Ts, (long)(DateTime.UtcNow - s_jwtBaselineTime).TotalSeconds ),
-                new JProperty(PoPClaimTypes.HttpMethod, _popAuthenticationConfiguration.HttpMethod?.ToString()),
-                new JProperty(PoPClaimTypes.Host, _popAuthenticationConfiguration.RequestUri.Host),
-                new JProperty(PoPClaimTypes.Path, _popAuthenticationConfiguration.RequestUri.AbsolutePath),
-                new JProperty(PoPClaimTypes.Nonce, CreateNonce()),
-                new JProperty(PoPClaimTypes.Cnf, new JObject(
-                    new JProperty(PoPClaimTypes.JWK, popAssertion))));
+            string popToken = CreateJWS(body.ToString(Formatting.None), header.ToString(Formatting.None));
+            return popToken;
+        }
 
-            string popToken =  CreateJWS(payload.ToString(Json.Formatting.None), header.ToString(Json.Formatting.None));
+        private JObject CreateBody(MsalAccessTokenCacheItem msalAccessTokenCacheItem)
+        {
+            JToken publicKeyJWK = JToken.Parse(_popAuthenticationConfiguration.PopCryptoProvider.CannonicalPublicKeyJwk);
+            List<JProperty> properties = new List<JProperty>(8);
```

Comment:
nit: Is there a reason this is set to a capacity of 8? Seems there aren't 8 properties atm and some may be null anyway. Not really a big deal though.

Reply:
It's a good question!

These higher level collections like Lists are usually backed up by arrays (List impl [here](https://referencesource.microsoft.com/#mscorlib/system/collections/generic/list.cs,cf7f4095e4de7646)). Array is of a fixed size, so it needs to be reallocated most likely using `Array.Copy`.

By default, a collections starts at capacity 0 (well, it dependes on .NET runtime, might be 4, but let's say 0). After you add 1 elements, it gets resized to 2. And then to 4, 8, 16 etc. Since we know the collection will have ~8 elements, I just set the capacity to 8 to avoid resizing. If we end up adding more than 8, we can set a more relevant capacity.